src/main/java/me/lucien/minesweeper/web/GameController.java

```diff
@@ -88,13 +86,29 @@ public int rightClick(@PathVariable("id") int id, @PathVariable("x") int x, @Pat
         if (x < 0 || x >= room.getWidth() || y < 0 || y >= room.getHeight()) {
             throw new IndexOutOfBoundsException();
         }
-
         room.flag(x, y);
+
         reTimer(id);
 
         return room.getBoard()[x][y].getState().ordinal();
     }
 
+    @PostMapping("room/{id}/batch")
+    public List<SquareData> doubleClick(@PathVariable("id") int id, @RequestParam("key") String key,
+                                        @RequestParam("x") int x, @RequestParam("y") int y) throws HttpException {
+        checkLegitimacy(id, key);
+
+        Room room = roomMap.get(id);
+        if (x < 0 || x >= room.getWidth() || y < 0 || y >= room.getHeight()) {
+            throw new IndexOutOfBoundsException();
+        }
+        List<SquareData> res = room.getAround(x, y);
```

Comment:
The method should uncover all the rest squares. So the verb here should not be `get`. How about `room.outspread(x, y)`